File: eCodes.Models/Requests/UserInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace eCodes.Models.Requests
{
    public class UserInsertRequest
    {
        //PersonInsertRequest
        [Required(AllowEmptyStrings = false)]
        public string FirstName { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string CityName { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string Jmbg { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string Gender { get; set; }
        //-----------------------------------------
        [Required(AllowEmptyStrings = false)]
        [EmailAddress()]
        public string Email { get; set; }
        [Required(AllowEmptyStrings = false)]
        [Phone()]
        public string PhoneNumber { get; set; }
        [MinLength(4)]
        [Required(AllowEmptyStrings = false)]
        public string Username { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string Password { get; set; }
        [Required(AllowEmptyStrings = false)]
        public string PasswordConfirmation { get; set; }
        public int PersonId { get; set; }
        public bool Status { get; set; }
        public List<int> UserRolesIdList { get; set; } = new List<int> { };
    }
}

File: ecodes_mobile/lib/model/requests/user_insert_request.dart
import 'package:json_annotation/json_annotation.dart';



part 'user_insert_request.g.dart';

@JsonSerializable()
class UserInsertRequest{

  String? firstName;
  String? lastName;
  DateTime? dateOfBirth;
  String? cityName;
  String? jmbg;
  String? gender;
  String? email;
  String? username;
  String? password;
  String? passwordConfirmation;

  UserInsertRequest(){

  }
factory UserInsertRequest.fromJson(Map<String, dynamic> json) => _$UserInsertRequestFromJson(json);

  /// Connect the generated [_$UserInsertRequestToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserInsertRequestToJson(this);


}
